DevWs for NIE-TECH CO., LTD. containing containing Evalogik Door/Window Sensor

sky-nie opened the pull request.

Winniewen18 committed:
DevWs for NIE-TECH CO., LTD. containing containing Evalogik Door/Window Sensor

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		fingerprint mfr: "0312", prod: "0713", model: "D100", deviceJoinName: "Minoston 3 in 1 Sensor" //MSE30Z/ZWS713
+	}
+
+	tiles(scale: 2) {
```
Tiles are not supported by ST app anymore. Please remove this section.

sky-nie replied:
OK, I remove it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		details(["contact", "temperature", "humidity", "battery", "refresh", "pendingChanges"])
+	}
+
+	simulator { }
```
Please remove this section.

sky-nie replied:
OK, I remove it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	}
+}
+
+private getOptionsInput(param) {
```
Since getOptionsInput and getNumberInput are called only once,  please put that code in the preferences section.

sky-nie replied:
OK, I move the function process to the place of call.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+			options: param.options
+}
+
+private getNumberInput(param) {
```
Since getOptionsInput and getNumberInput are called only once, please put that code in the preferences section.

sky-nie replied:
OK, I move the function process to the place of call.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+ */
+metadata {
+	definition(
+			name: "Evalogik Door/Window Sensor",
```
Could You indent this section with 2 tabs or put definition section in one line ?

sky-nie replied:
OK, I  adjust it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+			namespace: "sky-nie",
+			author: "winnie",
+			ocfDeviceType: "x.com.st.d.sensor.contact",
+			runLocally: true,
```
Does this DTH work in a full scope with `runLocally `flag set to true ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		if (state.refreshConfig) {
+			cmds << configGetCmd(param)
+		}
+		else if ("${storedVal}" != "${param.value}") {
```
Could You put closing bracket and "else if" statement in one line ?
`} else if {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+
+def executeConfigure() {
+	def cmds = [
+			sensorBinaryGetCmd(),
```
Could You indent this section with 2 tabs ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	def cmds = [
+			sensorBinaryGetCmd(),
+			batteryGetCmd(),
+			sensorMultilevelGetCmd(tempSensorType),
```
If You change minTemperatureOffsetParam or minHumidityOffsetParam - sensorMultilevelGetCmd will sent to the device twice because getCofingCmds will return that command too.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	state.lastBattery = null
+	if (!state.refreshSensors) {
+		state.refreshSensors = true
+	}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		}
+
+		sendEvent(name: "lastCheckIn", value: convertToLocalTimeString(new Date()), displayed: false)
+	}
```
Could You put closing bracket and "catch" statement in one line ?
`} catch {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	def result = []
+	if (encapCmd) {
+		result += zwaveEvent(encapCmd)
+	}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	if (state.refreshSensors) {
+		cmds += [
+				sensorBinaryGetCmd(),
+				sensorMultilevelGetCmd(tempSensorType),
```
If You change minTemperatureOffsetParam or minHumidityOffsetParam and device wakes up then  sensorMultilevelGetCmd will sent to the device twice because getCofingCmds will return that command too.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	}
```
Could You put closing bracket and "else if" statement in one line ?
`} else if {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	logTrace "SensorMultilevelReport: ${cmd}"
+
+	if (cmd.sensorValue != [255, 255]) { // Bug in beta device
```
If there is a bug in beta device, will it be fixed in production firmware ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+
+				sendEvent(getEventMap("temperature", temp, true, null, getTemperatureScale()))
+				break
+
```
Could You remove this empty line ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+
+		logDebug "${param.name}(#${param.num}) = ${val}"
+		setParamStoredValue(param.num, val)
+	}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	logTrace "NotificationReport: $cmd"
+	def result = []
+
+	if (cmd.notificationType == 0x06 && cmd.event == 0x16) {
```
Could You use constants for notification types and events ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	def isStateChange = (device.currentValue(name) != value)
+	displayed = (displayed == null ? isStateChange : displayed)
+	def eventMap = [
+			name: name,
```
Could You indent this section with 2 tabs ?

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	map.name = "contact"
+	if (map.value == "open") {
+		map.descriptionText = "${device.displayName} is open"
+	}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+	try {
+		if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
+			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		}
```
Could You put closing bracket and "else" statement in one line ?
`} else {`

sky-nie replied:
OK, I modify all  "else" at the end of closing bracket.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+		else {
+			return cmd.format()
+		}
+	}catch (ex) {
```
Nit: add space before `catch`

sky-nie replied:
OK, I modify it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+
+private static getCommandClassVersions() {
+	[
+			0x30: 2,  // SensorBinary
```
Could You indent this section with 2 tabs ?

sky-nie replied:
OK, I  adjust it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+// Configuration Parameters
+private getConfigParams() {
+	[
+			batteryReportThresholdParam,
```
Could You indent this section with 2 tabs ?

sky-nie replied:
OK, I adjust it.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,683 @@
+
+private getAssociationGroupSettingParam() {
+	return getParam(13, "Association Group 2 Setting", 1, 1, [0:"Disable completely",
+															  1:"Send Basic SET 0xFF when Magnet is away,and send Basic SET 0x00 when Magnet is near.",
```
Could You reformat this section ?

sky-nie replied:
OK, I  adjust it.

Winniewen18 committed:
1.Syntax format compliance adjustment
2.delete dummy code

Winniewen18 committed:
Syntax format compliance adjustment

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+						required: false,
+						defaultValue: "${it.value}",
+						range: it.range
+				}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+ *
+ */
+metadata {
+	definition(name: "Evalogik Door/Window Sensor",namespace: "sky-nie",author: "winnie",ocfDeviceType: "x.com.st.d.sensor.contact",runLocally: true,executeCommandsLocally: false,genericHandler: "Z-Wave") {
```
Nit: add space after commas

KKlimczukS replied:
Does this DTH work in a full scope with runLocally flag set to true ?

sky-nie replied:
I removed runLocally  as IDE required. It seems to be no different from before.

sky-nie replied:
I removed runLocally  as IDE required. It seems to be no different from before.

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+		def storedVal = getParamStoredValue(param.num)
+		if (state.refreshConfig) {
+			cmds << configGetCmd(param)
+		}else if ("${storedVal}" != "${param.value}") {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+	state.lastBattery = null
+	if (!state.refreshSensors) {
+		state.refreshSensors = true
+	}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+		def cmd = zwave.parse(description, commandClassVersions)
+		if (cmd) {
+			result += zwaveEvent(cmd)
+		}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+	def result = []
+	if (encapCmd) {
+		result += zwaveEvent(encapCmd)
+	}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+	def val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	}else if (val < 1) {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+
+		logDebug "${param.name}(#${param.num}) = ${val}"
+		setParamStoredValue(param.num, val)
+	}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+	map.name = "contact"
+	if (map.value == "open") {
+		map.descriptionText = "${device.displayName} is open"
+	}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+	try {
+		if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
+			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		}else {
```
Nit: spacing

KKlimczukS commented on devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy:
```diff
@@ -0,0 +1,616 @@
+			return cmd.format()
+		}
+	} catch (ex) {
+		return cmd.format()
```
Why are You returning this `cmd.format()`  here? This is a block for handling exceptions ?

sky-nie replied:
I found a crash when I run a virtual device in this function, And I repaired it when I insert "return cmd.format()" at the catch branch.

KKlimczukS replied:
What's the root cause of that crash ? Have You tested this DTH with a real device ?

sky-nie replied:
No crash happened when I test with a real device.

sky-nie replied:
The function secureCmd was previously implemented like this:

`if (zwaveInfo.zw.contains("s") || ("0x98" in device.rawDescription.split(" "))) {
			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
		} else {
			return cmd.format()
		}`

When I run a virtual device in IDE, crash hanppened in this function, at line "if (zwaveInfo.zw.contains("s") || ("0x98" in device.rawDescription.split(" "))) {";
I change it to "if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {", The crash still happened.
The IED can not tell me more about the carash, I change function like below :

`private secureCmd(cmd) {
	try {
		if (zwaveInfo?.zw?.contains("s") || ("0x98" in device?.rawDescription?.split(" "))) {
			return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
		} else {
			return cmd.format()
		}
	} catch (ex) {
		return cmd.format()
	}
}`

Then the carsh disappear. No crash happened when a real device run with this DTH.

KKlimczukS replied:
That's because virtual devices might not have zwaveInfo details. 
DTH is dedicated for real devices, so this catch(ex) block is not necessary. 

If You need it **for development purposes** You might use this code in catch(e) block to see details in the live logging.
`throw new RuntimeException(e)`

sky-nie replied:
Thanks, I understand, I know how to do it.

Winniewen18 committed:
Syntax format compliance adjustment

Winniewen18 committed:
1. update ConfigParams as product designed
2.update DTH name  from "Evalogik Door/Window Sensor" to "Minoston Door/Window Sensor"

Winniewen18 committed:
1.change lastBatteryReport to record the time of fresh battery
2.add lastBattery to record the battery value
3.adjust the preferences content for ST

KKlimczukS commented on devicetypes/sky-nie/minoston-door-window-sensor.src/minoston-door-window-sensor.groovy:
```diff
@@ -0,0 +1,674 @@
+ *
+ */
+metadata {
+	definition(name: "Minoston Door/Window Sensor", namespace: "sky-nie", author: "winnie", ocfDeviceType: "x.com.st.d.sensor.contact", executeCommandsLocally: false, genericHandler: "Z-Wave") {
```
@Winniewen18 Why did You add another DTH to this pull request ? That should be in a separate CST request.

sky-nie replied:
Oh, Something similar happened before.This is caused by upgrading and modifying the name.Not another DTH:
  
![1626420119_副本](https://user-images.githubusercontent.com/54890556/125908673-35f391dd-ac5e-42b4-9a2a-2cb7c7c7bcec.jpg)
   Because the products related to the brand “Evalogik” have been suspended, this DTH has nothing to do with  “Evalogik”. In order not to cause confusion and misunderstanding, it was renamed.

KKlimczukS replied:
@sky-nie that's not just renaming the current DTH. 
Another code (completely different DTH) was submitted. Please leave in this PR only the DTH that was reviewed and rename it if it's needed.

sky-nie replied:
@KKlimczukS :
   Before I renamed "Evalogik Door/Window Sensor"  to "Minoston Door/Window Sensor" to this DTH, There was another DTH with name of "Minoston Door/Window Sensor" in my fork(which is not related with Minoston Door/Window Sensor ).  I deleted it before the rename action. 
   What can I do now? restore the current DTH's name?

KKlimczukS replied:
@sky-nie 
Can You try to 
- revert DTH's (the one that was reviewed) name to the previous one and commit
- delete DTH (the one that is unnecessary) and commit
- change the name of the DTH again and commit

sky-nie replied:
I have completed the first step. The second step,,, now I can not see the unnecessary DTH. Is the state back to normal? The third step——I think renaming should not be a necessary action, it should be ok to add enough comments in the file.

KKlimczukS replied:
This unnecessary DTH is still here in this PR, so You need to have Your local branch and this PR synchronized.

sky-nie replied:
I have  sync the local branch with my fork, update the locale branch , then commit  the locale branch to [this PR](https://github.com/sky-nie/SmartThingsPublic/tree/devws-for-nie-tech-co-ltd-contai_51);
If it still doesn't work, I have to close the PR  and start  a new PR;.@KKlimczukS

KKlimczukS replied:
@sky-nie Please create a fresh branch and issue new pull request (and add a link to it in this PR).

sky-nie replied:
@KKlimczukS :
   I have create a fresh branch and issue new pull request( [the PR](https://github.com/sky-nie/SmartThingsPublic/tree/devws-for-nie-tech-co-ltd-contai_53)) .

Winniewen18 committed:
restore name to old

Winniewen18 committed:
sync local branch with my fork, update the last to locale, then pull tlocale o the pull request branch.

KKlimczukS commented:
@sky-nie Since You submitted new PR #71706 (Evalogik Door/Window Sensor), can this PR be closed now ?

sky-nie commented:
A new PR about this project restart.
